src/status_im/wallet_connect/core.cljs

```diff
@@ -0,0 +1,277 @@
+
+(fx/defn update-sessions
+  {:events [:wallet-connect/update-sessions]}
+  [{:keys [db] :as cofx}]
+  (let [client (get db :wallet-connect/client)]
+    {:db (-> db
+             (assoc :wallet-connect/sessions (js->clj (.-values (.-session client)) :keywordize-keys true))
+             (dissoc :wallet-connect/session-managed))}))
+
+(fx/defn wallet-connect-complete-transaction
+  {:events [:wallet-connect.dapp/transaction-on-result]}
+  [{:keys [db]} message-id topic result]
+  (let [client (get db :wallet-connect/client)
+        response {:topic topic
+                  :response {:jsonrpc "2.0"
+                             :id message-id
+                             :result result}}]
+    {:db (assoc db :wallet-connect/response response)
+     :wc-2-respond [client response]}))
+
+(fx/defn wallet-connect-send-async
+  [cofx {:keys [method params id] :as payload} message-id topic]
+  (let [message?      (browser/web3-sign-message? method)
+        dapps-address (get-in cofx [:db :multiaccount :dapps-address])
+        accounts (get-in cofx [:db :multiaccount/visible-accounts])
+        typed? (and (not= constants/web3-personal-sign method) (not= constants/web3-eth-sign method))]
+    (if (or message? (= constants/web3-send-transaction method))
+      (let [[address data] (cond (and (= method constants/web3-keycard-sign-typed-data)
+                                      (not (vector? params)))
+                                 ;; We don't use signer argument for keycard sign-typed-data
+                                 ["0x0" params]
+                                 message? (browser/normalize-sign-message-params params typed?)
+                                 :else [nil nil])]
+        (when (or (not message?) (and address data))
+          (signing/sign cofx (merge
+                              (if message?
+                                {:message {:address address
+                                           :data data
+                                           :v4 (= constants/web3-sign-typed-data-v4 method)
+                                           :typed? typed?
+                                           :pinless? (= method constants/web3-keycard-sign-typed-data)
+                                           :from address}}
+                                {:tx-obj  (-> params
+                                              first
+                                              (update :from #(or % dapps-address))
+                                              (dissoc :gasPrice))})
+                              {:on-result [:wallet-connect.dapp/transaction-on-result message-id topic]
+                               :on-error  [:wallet-connect.dapp/transaction-on-error message-id topic]}))))
+      (when (#{"eth_accounts" "eth_coinbase"} method)
+        (wallet-connect-complete-transaction cofx message-id topic (if (= method "eth_coinbase") dapps-address [dapps-address]))))))
+
+(def permissioned-method
```

Comment:
same methods as for dapps ? why copy them ?

Reply:
I think is not used, will delete it